cmd/capibmadm/cmd/powervs/network/create.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package network
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/options"
+)
+
+type networkCreateOptions struct {
+	name       string
+	dnsServers []string
+}
+
+// NewNetworkCreateCommand function to create PowerVS network.
+func NewNetworkCreateCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "network-create NETWORK_NAME",
+		Aliases: []string{"netc"},
+		Short:   "Create PowerVS network",
+		Example: `
+# Create PowerVS network with name capi-network
+capibm powervs network-create capi-network --service-instance-id <servcie-instance-id> --api-key <api-key>
+
+# Create PowerVS network with name capi-network by setting environmental variables
+export IBMCLOUD_API_KEY=<api-key>
+export SERVICE_INSTANCE_ID=<servcie-instance-id>
+capibm powervs network-create capi-network`,
+	}
+
+	var netCreateOption networkCreateOptions
+	cmd.Flags().StringSliceVar(&netCreateOption.dnsServers, "dns-servers", []string{"8.8.8.8", "9.9.9.9"}, "Comma separated list of DNS Servers to use for this network")
+
+	cmd.RunE = func(cmd *cobra.Command, args []string) error {
+		if len(args) < 1 {
+			return fmt.Errorf("network name is not provided")
+		}
+		netCreateOption.name = args[0]
+
+		if err := createNetwork(netCreateOption); err != nil {
+			return err
+		}
+		return nil
+	}
+	return cmd
+}
+
+func createNetwork(netCreateOption networkCreateOptions) error {
+	fmt.Printf("Creating Power VS network with name: %s, service instance id: %s, dns-servers %s\n", netCreateOption.name, options.ServiceInstanceID, netCreateOption.dnsServers)
```

Comment:
Here I have just used fmt.Printf just to validate everything is working as expected this can be removed during actual implementation but when comes to logging should be we use klog or the logs used by clusterctl in capi: https://github.com/kubernetes-sigs/cluster-api/blob/3b4953a9021de8abf184b5fba59ff43e36738667/cmd/clusterctl/log/log.go#L32

Reply:
as we discussed lets look at the available options like logr vs klog

Reply:
I have updated it use logr, which is being used by clusterctl and provided ability to have key value pairs in logs.